File: pm_core_1/src/infrastructure/store/model/label_model.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package model

type LabelModel struct {
	BaseModel

	ProjectID int64  `gorm:"column:project_id;not null;index:idx_label_project"`
	Name      string `gorm:"column:name;type:varchar(100);not null"`
	Color     string `gorm:"column:color;type:varchar(20);not null"`
}

func (LabelModel) TableName() string {
	return "labels"
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/model/LabelModel.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "labels")
@SQLRestriction("deleted_at IS NULL")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@SuperBuilder
public class LabelModel extends BaseModel {

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "project_id", nullable = false)
    private Long projectId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "color")
    private String color;
}
